network/proxy.go

```diff
@@ -553,9 +553,14 @@ func (pr *Proxy) PassThroughToClient(conn *ConnWrapper, stack *Stack) *gerr.Gate
 		span.AddEvent("Plugin(s) modified the response")
 	}
 
-	// Send the response to the client.
-	errVerdict := pr.sendTrafficToClient(conn.Conn(), response, received)
-	span.AddEvent("Sent traffic to client")
+	var errVerdict *gerr.GatewayDError
+	if received > 0 {
+		// Send the response to the client.
+		errVerdict = pr.sendTrafficToClient(conn.Conn(), response, received)
```

Comment:
I’m not sure we need to run the OnTrafficToClient hooks if sendTrafficToClient fails. If we do need it, we can handle this block like this:
```		
	if received > 0 {
		errVerdict = pr.sendTrafficToClient(conn.Conn(), response, received)
		if errVerdict != nil {
			span.RecordError(errVerdict)
			span.AddEvent("Failed to send traffic to client")
		} else {
			span.AddEvent(fmt.Sprintf("Sent %d bytes of traffic to client", received))
		}
	} else {
		span.AddEvent("No data to send to client")
	}
	
```
If we don’t need it, I think we can change the logic like this:
```
	if received == 0 {
		span.AddEvent("No data to send to client")
		return nil
	}

	// Send the response to the client.
	errVerdict := pr.sendTrafficToClient(conn.Conn(), response, received)
	if errVerdict != nil {
		span.RecordError(errVerdict)
		span.AddEvent("Failed to send traffic to client")
		return errVerdict
	}

	span.AddEvent(fmt.Sprintf("Sent %d bytes of traffic to client", received))
```

Reply:
Addressed in https://github.com/gatewayd-io/gatewayd/pull/610/commits/15c3ad0cee46d91410b81e169c611c54f03eaf20.